project-koku/koku
# Azure Local Download/Processing

Azure-Local download works in a similar way to AWS-local.  The following provider payload is an example of what to set when creating the provider

```
{"name": "Test Azure Source", "type": "AZURE-local", "authentication": {
        "credentials": {
            "subscription_id": "329cc628-23e6-45f1-9993-736ea5c79216",
            "tenant_id": "91910d45-f47a-4dc2-a667-39dd7ab0fd65",
            "client_id": "85dbba8e-5e3c-4f42-97c3-e58a5492dbca",
            "client_secret": "ahhhhh"
        }
    }, "billing_source": {
        "data_source": {"resource_group": {"export_name": "myexport", "directory": "dir"},
        	"storage_account": {"local_dir": "/tmp/local_container", "container": "my_container"}
        }
    }
}
```

The `local_dir` directory is equivalent to the Azure "container". If using docker-compose this must be mapped to a directory on your local host.  For example, from my test results file I have this mounted to `~/azure_local`.

Nise is in the process of being updated to support Azure reports.  The reports generated in the unit test were done using https://github.com/project-koku/nise/tree/azure_data_generation/nise

**Testing**
1. Regression test the real Azure Workflow
2. Create an Azure local container on host machine with directory, export name, and date range
3. Generate multiple Azure Reports with nise and copy them to this local directory
4. Add an AZURE-local provider with the following payload
5. Verify that the docker mount maps from the local container directory to the `local_dir` path in the data_source.
6. Verify that the most recent csv file is in the /var/tmp/masu directory


**Test Results**
[azure_local_downloader_ut.txt](https://github.com/project-koku/koku/files/3555529/azure_local_downloader_ut.txt)
